File: code_examples/3-Structured_Problem_Solving/15-For_Loop/C/main.c
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program uses a for loop

#include <stdio.h>

int main() {
    // this function uses a for loop
    int positiveInteger;

    // input
    printf("Enter how many times to repeat: ");
    scanf("%d", &positiveInteger);
    printf("\n");

    // process & output
    for (int loopCounter = 0; loopCounter < positiveInteger; loopCounter++) {
        printf("%d time(s) through the loop.\n", loopCounter);
    }

    printf("\nDone.\n");
    return 0;
}

File: code_examples/3-Structured_Problem_Solving/15-For_Loop/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module uses a for loop
"""


def main() -> None:
    """The main() function uses a for loop, returns None."""

    # input
    positive_integer = int(input("Enter how many times to repeat: "))
    print("")

    # process & output
    for loop_counter in range(positive_integer):
        print(f"{loop_counter} time through loop.")

    print("\nDone.")


if __name__ == "__main__":
    main()
